src/Microsoft.AspNet.Hosting/HostingServices.cs

```diff
@@ -25,28 +26,34 @@ private static IServiceCollection Import(IServiceProvider fallbackProvider)
             return services;
         }
 
-        public static IServiceCollection Create(IConfiguration configuration = null)
+        // REVIEW: should additionalHostServices be an IServiceProvider??
+        public static IServiceCollection Create(IConfiguration configuration = null, IServiceCollection additionalHostServices = null)
         {
-            return Create(CallContextServiceLocator.Locator.ServiceProvider, configuration);
+            return Create(CallContextServiceLocator.Locator.ServiceProvider, configuration, additionalHostServices);
         }
 
-        public static IServiceCollection Create(IServiceProvider fallbackServices, IConfiguration configuration = null)
+        public static IServiceCollection Create(IServiceProvider fallbackServices, IConfiguration configuration = null, IServiceCollection additionalHostServices = null)
         {
             configuration = configuration ?? new Configuration();
-            var services = Import(fallbackServices);
+            var services = Import(fallbackServices, additionalHostServices);
             services.AddHosting(configuration);
-            services.AddSingleton<IServiceManifest>(sp => new HostingManifest(fallbackServices));
+            services.AddSingleton<IServiceManifest>(sp => new HostingManifest(fallbackServices, additionalHostServices));
```

Comment:
@davidfowl The code here, is passing the collection to the manifest

The complication with making it an Action, is that we need to be able to figure out what services were added so we add them to the manfiest.  If we take the ServiceCollection itself rather than the action, we know that every service in there should be part of the manivest.  If we simply run the action as part of the service collection we are building, we have no way to determine which services came from hosting (but shouldn't be exported), vs ones they really want added.